src/core/dev_api/openvino/runtime/shared_buffer.hpp

```diff
@@ -28,4 +28,19 @@ class SharedBuffer : public ov::AlignedBuffer {
     T _shared_object;
 };
 
+
+/// \brief SharedStreamBuffer class to store pointer to pre-acclocated buffer and provide streambuf interface.
+template <typename T>
+class SharedStreamBuffer final : public std::stringbuf {
+public:
+    SharedStreamBuffer(char* data, size_t size, const T& shared_object) : _shared_object(shared_object) {
+        basic_streambuf::pubsetbuf(data, size);
+    }
+
+    SharedStreamBuffer() = delete;
+
+private:
+    T _shared_object;
```

Comment:
Maybe this class has to not be so generic and `shared object` can be stored as shared pointer to void. 
example
```cpp
class SharedStreamBuffer final : public std::stringbuf {
public:
    SharedStreamBuffer(char* data, size_t size, std::shared_ptr<void> shared_object) : m_shared_object(std::move(shared_object)) {
        basic_streambuf::pubsetbuf(data, size);
    }

    SharedStreamBuffer() = delete;

private:
    std::shared_ptr<void> m_shared_object;
};
```

Reply:
I change it. Now it creates (in RAII manner) SharedBuffer, which already has an implementation to own the shared object